### PR TITLE
Add prompt for settings file of existing Django project

### PR DESCRIPTION
**Pull request by wangxf123456**

### CODE REVIEW

File: django_cloud_deploy/cli/prompt.py [hunk 1]
```diff
@@ -1136,6 +1138,81 @@ def _validate(self, s: str):
                               '"test@example.com"').format(s))
 
 
+class DjangoSettingsPathPrompt(StringTemplatePrompt):
+    """Allow the user to enter the settings file path of a Django project."""
+
+    PARAMETER = 'django_settings_path'
+    MESSAGE = ('{} Enter the settings file path you want to use for deployment')
```
Comment (brianquinlan) [inline django_cloud_deploy/cli/prompt.py]: This isn't completely clear to me. Maybe explain it and we can work on the wording a bi t?

File: django_cloud_deploy/cli/prompt.py [hunk 2]
```diff
@@ -1136,6 +1138,81 @@ def _validate(self, s: str):
...
+
+        if not django_settings_path.endswith('.py'):
+            raise ValueError(
+                '["{}"] is not a .py file.'.format(django_settings_path))
+
+        module_relative_path = os.path.relpath(django_settings_path,
+                                               project_dir)
+
+        # Remove ".py" at the end.
+        module_name = module_relative_path[:-3].replace('/', '.')
```
Comment (brianquinlan) [inline django_cloud_deploy/cli/prompt.py]: os.path.splitext will remove the .py for you - and you can use it for the extension checking above too

File: django_cloud_deploy/cli/prompt.py [hunk 3]
```diff
@@ -1136,6 +1138,81 @@ def _validate(self, s: str):
...
+            args: Dictionary holding prompts answered by user and set up
+                command-line arguments.
+
+        Returns:
+            A Copy of args + the new parameter collected.
+        """
+        new_args = dict(args)
+
+        django_directory_path = args.get('django_directory_path', None)
+        validate = functools.partial(self._validate, django_directory_path)
```
Comment (brianquinlan) [inline django_cloud_deploy/cli/prompt.py]: Move this right before _ask_prompt

Comment (wangxf123456) [reply django_cloud_deploy/cli/prompt.py]: Actually it is used on the next line
```python
if self._is_valid_passed_arg(console, step, args.get(self.PARAMETER, None), validate):
```

File: django_cloud_deploy/cli/prompt.py (hunk 2 above)
Comment (wangxf123456) [reply django_cloud_deploy/cli/prompt.py]: Done.

File: django_cloud_deploy/cli/prompt.py (hunk 3 above)
Comment (brianquinlan) [reply django_cloud_deploy/cli/prompt.py]: Ahhh, thanks.

File: django_cloud_deploy/cli/prompt.py (hunk 1 above)
Comment (wangxf123456) [reply django_cloud_deploy/cli/prompt.py]: I added more comments for this class.

Comment (brianquinlan) [reply django_cloud_deploy/cli/prompt.py]: I mean that the message to the user isn't completely clear. I can't think of anything much better:

```suggestion
    MESSAGE = ('{} Enter the path of the Django settings file that should be used for deployment')
```

Comment (wangxf123456) [reply django_cloud_deploy/cli/prompt.py]: Done.

File: django_cloud_deploy/cli/prompt.py [hunk 4]
```diff
@@ -1136,6 +1138,89 @@ def _validate(self, s: str):
...
+
+        Returns:
+            A Copy of args + the new parameter collected.
+        """
+        new_args = dict(args)
+
+        django_directory_path = args.get('django_directory_path_cloudify', None)
+        validate = functools.partial(self._validate, django_directory_path)
+        if self._is_valid_passed_arg(console, step,
+                                     args.get(self.PARAMETER, None), validate):
```
Comment (brianquinlan) [inline django_cloud_deploy/cli/prompt.py]:
```suggestion
                                     args.get(self.PARAMETER), validate):
```

**Pull request merged**